backend/reservation-service/src/main/kotlin/com/ticketqueue/reservation/service/ReservationService.kt

```
@@ -138,61 +127,148 @@ class ReservationService(
+        }
+    }
+
+    /**
+     * 좌석 변경 서비스 (REQ-RSV-002)
+     *
+     * ## 변경 프로세스
+     * 1. 기존 + 신규 좌석 전체를 정렬하여 MultiLock 획득 — 데드락 방지
+     * 2. 신규 좌석(seatsToAcquire)만 hold_seats SET 중복 확인 — 기존 좌석 오탐 방지
+     * 3. DB 트랜잭션 + afterCommit 콜백으로 Redis hold_seats SET 갱신 — DB 롤백 시 Redis 오염 없음
+     */
+    fun changeSeats(userId: UUID, reservationId: UUID, request: ChangeSeatsRequest, queueToken: String): ChangeSeatsResponse {
```

detekt (1.23.8)

[warning] The function changeSeats appears to be too complex based on Cognitive Complexity (complexity: 27). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)